packages/test/src/test-custom-data-converter.ts

```diff
@@ -0,0 +1,139 @@
+import { Connection, WorkflowClient } from '@temporalio/client';
+import { coresdk } from '@temporalio/proto';
+import { Core, DefaultLogger, Worker } from '@temporalio/worker';
+import test, { ExecutionContext } from 'ava';
+import { v4 as uuid4 } from 'uuid';
+import { protoActivity } from './activities';
+import { dataConverter, messageInstance } from './data-converters/data-converter';
+import { cleanStackTrace, RUN_INTEGRATION_TESTS } from './helpers';
+import { defaultOptions, isolateFreeWorker } from './mock-native-worker';
+import { protobufWorkflow } from './workflows';
+
+export async function runWorker(worker: Worker, fn: () => Promise<any>): Promise<void> {
+  const promise = worker.run();
+  await fn();
+  worker.shutdown();
+  await promise;
+}
+
+function compareCompletion(
+  t: ExecutionContext,
+  actual: coresdk.activity_result.IActivityResult | null | undefined,
+  expected: coresdk.activity_result.IActivityResult
+) {
+  if (actual?.failed?.failure) {
+    const { stackTrace, ...rest } = actual.failed.failure;
+    actual = { failed: { failure: { stackTrace: cleanStackTrace(stackTrace), ...rest } } };
+  }
+  t.deepEqual(
+    new coresdk.activity_result.ActivityResult(actual ?? undefined).toJSON(),
+    new coresdk.activity_result.ActivityResult(expected).toJSON()
```

Comment:
I thought you're supposed to use `create`

Reply:
For json-modules, yes. These are coresdk generated classes (`compareCompletion()` is copied from other test files).